keysync_shared.go

```diff
@@ -0,0 +1,75 @@
+package nitriding
+
+import (
+	cryptoRand "crypto/rand"
+	"encoding/base64"
+	"errors"
+	"time"
+
+	"golang.org/x/crypto/nacl/box"
+)
+
+const (
+	boxKeyLen = 32 // NaCl box's private and public key length.
+)
+
+var (
+	// Instead of using rand.Read or time.Now directly, we use the following
+	// variables to enable mocking as part of our unit tets.
+	cryptoRead  = cryptoRand.Read
+	currentTime = func() time.Time { return time.Now().UTC() }
+)
+
+// nonce represents a nonce that's used to prove the freshness of an enclave's
+// attestation document.
+type nonce [nonceLen]byte
+
+// boxKey represents key material for NaCl's box, i.e., a private and a public
+// key.
+type boxKey struct {
+	pubKey  *[boxKeyLen]byte
+	privKey *[boxKeyLen]byte
+}
```

Comment:
I'm not convinced it makes sense to have a struct here. Because it's using SealAnonymous, the responder never needs any private keys. Meanwhile the .Bytes() and newBoxKeyFromBytes() functions are misleading because they only deal with the public half of the keys.

I'd suggest defining just `type boxKey [boxKeyLen]byte`. Then change `newBoxKeyFromBytes` to return only the half of the keypair it's shown. Inline `newBoxKey` at the call site in keysync_initiator and just track the pubKey and privKey in separate variables. Or make it return the pair if your like that better for the unit test call sites.

Reply:
Agreed. As mentioned above, fixed in 493fad2. I kept the struct, so we don't have to pass yet another argument to `processAttDoc` and to lessen the possible confusion of passing three types of "keys".